src/sentry/integrations/gitlab/blame.py

```diff
@@ -83,9 +83,10 @@ def _fetch_file_blame(
     project_id = file.repo.config.get("project_id")
     encoded_path = quote(file.path, safe="")
     request_path = GitLabApiClientPath.blame.format(project=project_id, path=encoded_path)
-    response = client.get(
```

Comment:
Is there a good way for us to be able to record to a log (and perhaps a metric as well) when the cache has been used? One thing I'm worried about is the cache either 1. never being used or 2. being used a lot and causing bugs that will be difficult to track down.

Sorry for not specifying in the original issue, we can add in a followup, just wanted to start thinking about how we might do that

Reply:
I think just right [here](https://github.com/getsentry/sentry/blob/master/src/sentry/shared_integrations/client/base.py#L344) is when the cache is hit, we could log with the cache key and add metrics with just the integration name and when it's hit.

Reply:
That's true, but I think we will want more information than just the integration name, since this integration is used for multiple purposes. Maybe we could add a `check_cache(path, query, data)` function? That way we could log anything we want in `get_blame_for_files` if we see something there already.